src/robos.py

```
@@ -134,56 +134,66 @@ def sense_act(self):
+                        with battery_lock:  # Garantindo exclusividade na coleta da bateria
+                            if self.grid.get_snapshot()[target_pos[1]][target_pos[0]] == '⚡':  # Verifica novamente a bateria
+                                self.grid.clear_cell(target_pos)  # Remove a bateria do grid
+                                with self.locks['robots_mutex']:
+                                    # Atualiza a energia, limitada a 100
+                                    self.E = min(100, self.E + 20)
+                                    self.robots_info[self.id]['E'] = self.E
+                                self.logger.info(f"Robo {self.id} coletou bateria em {target_pos}. Energia: {self.E}")
+                            else:
+                                self.logger.info(f"Bateria em {target_pos} já foi coletada por outro robô.")
+                
+                elif cell_content != ' ' and cell_content != '⚡' and cell_content != '#':
```

Pylint (3.3.7)

[refactor] Consider merging these comparisons with 'in' by using 'cell_content not in (' ', '⚡', '#')'. Use a set instead if elements are hashable.

(R1714)